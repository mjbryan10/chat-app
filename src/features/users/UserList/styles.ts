import styled from 'styled-components';

export const ListWrapper = styled.ul`
   width: 100%;
   list-style: none;
   margin: 0;
   padding: 0;
   & li {
      width: 100%;
      margin-bottom: 5px;
      &:first-of-type {
         margin-top: 5px;
      }
   }
`;
