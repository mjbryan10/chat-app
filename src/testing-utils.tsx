import React from 'react';
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { dark } from './shared/theme';

const AllTheProviders = ({ children }: any) => {
  return (
    <ThemeProvider theme={dark}>
        {children}
    </ThemeProvider>
  )
  
}

const customRender = (ui?: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }