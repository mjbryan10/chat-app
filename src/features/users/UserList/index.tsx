import React, { FC, useEffect, useState } from 'react';
import { User } from 'shared/Api/types';
import UserItem from '../UserItem';
import * as S from './styles';

interface Props {
   users: User[];
   handleItemClick?: (user: User) => void;
}

const UserList: FC<Props> = ({ users, handleItemClick }) => {
   const onItemClick = (user: User) => {
      if (handleItemClick) handleItemClick(user);
   };
   const [usersState, setUsersState] = useState(users);
   useEffect(() => {
      setUsersState(users)
   }, [users])
   return (
      <S.ListWrapper>
         {usersState.map((user) => (
            <li key={user.id+'li'}>
               <UserItem key={user.id} user={user} handleClick={onItemClick} />
            </li>
         ))}
      </S.ListWrapper>
   );
};

export default UserList;
