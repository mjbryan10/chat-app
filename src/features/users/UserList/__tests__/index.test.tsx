import React from 'react';
import UserList from '..';
import { RenderResult, render, cleanup } from 'testing-utils';
import { fireEvent } from '@testing-library/react';

describe('UsersList Feature Component', () => {
   const __users = [
      {
         id: 1,
         name: 'Wessel',
      },
      {
         id: 2,
         name: 'Quint',
      },
      {
         id: 3,
         name: 'Mani',
      },
   ];
   const __handleItemClick = jest.fn();
   let utils: RenderResult;
   beforeEach(
      () =>
         (utils = render(
            <UserList users={__users} handleItemClick={__handleItemClick} />
         ))
   );
   afterEach(() => {
      cleanup();
   });

   test('should display the names of users given', () => {
    const {getByText} = utils;

      expect(getByText(/wessel/i)).toBeVisible();
      expect(getByText(/quint/i)).toBeVisible();
      expect(getByText(/mani/i)).toBeVisible();
   });

   test('should call click handler with id of user', () => {
    const {getAllByTestId} = utils;
    const firstUserItem = getAllByTestId('user-item')[0];

    fireEvent.click(firstUserItem);

      expect(__handleItemClick).toBeCalledTimes(1);
      expect(__handleItemClick).toBeCalledWith(__users[0]);
   });
});
