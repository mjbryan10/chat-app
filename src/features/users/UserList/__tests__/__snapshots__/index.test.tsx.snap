// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList Feature Component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="sc-AxirZ kPHIsu"
  >
    <li>
      <div
        class="sc-AxjAm fNKPyA"
        data-testid="user-item"
      >
        Wessel
      </div>
    </li>
    <li>
      <div
        class="sc-AxjAm fNKPyA"
        data-testid="user-item"
      >
        Quint
      </div>
    </li>
    <li>
      <div
        class="sc-AxjAm fNKPyA"
        data-testid="user-item"
      >
        Mani
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
